packages/opentelemetry-sdk-trace-base/src/export/BatchSpanProcessorBase.ts

```diff
@@ -166,6 +168,17 @@ export abstract class BatchSpanProcessorBase<T extends BufferConfig> implements
             }
           }
         );
+
+        const pendingResources = spans.map(span => span.resource)
+          .filter(resource => !resource.asyncAttributesHaveResolved());
+
+        // Avoid scheduling a promise to make the behavior more predictable and easier to test
+        if (pendingResources.length === 0) {
+          doExport();
+        } else {
+          Promise.all(pendingResources.map(resource => resource.waitForAsyncAttributes()))
```

Comment:
If resource detection is slower then `_exportTimeoutMillis` we may end up in exporting spans even the scheduled export has already timed out.
Usually parallel exports are not done by BatchSpanProcessor.